cmd2/parsing.py

```diff
@@ -247,23 +296,32 @@ def __init__(
         self._command_pattern = re.compile(expr)
 
     def is_valid_command(self, word: str) -> Tuple[bool, str]:
-        """Determine whether a word is a valid alias.
+        """Determine whether a word is a valid name for a command.
 
-        Aliases can not include redirection characters, whitespace,
-        or termination characters.
+        Commands can not include redirection characters, whitespace,
+        or termination characters. They also cannot start with a
+        shortcut.
 
-        If word is not a valid command, return False and a comma
-        separated string of characters that can not appear in a command.
+        If word is not a valid command, return False and error text
         This string is suitable for inclusion in an error message of your
         choice:
 
-        valid, invalidchars = statement_parser.is_valid_command('>')
+        valid, errmsg = statement_parser.is_valid_command('>')
         if not valid:
-            errmsg = "Aliases can not contain: {}".format(invalidchars)
+            errmsg = "Alias {}".format(errmsg)
         """
         valid = False
 
-        errmsg = 'whitespace, quotes, '
+        if not word:
+            return False, 'cannot be an empty string'
+
+        for (shortcut, expansion) in self.shortcuts:
+            if word.startswith(shortcut):
+                errmsg = 'cannot start with a shortcut: '
+                errmsg += ', '.join(shortcut for (shortcut, expansion) in self.shortcuts)
```

Comment:
What is this line supposed to do?  It is confusing and doesn't make sense to me.  We are already inside a for loop that loops over (shortcut, expansion) in self.shortcuts.

Reply:
The lines to build `errmsg` used to be before the loop, but that assignment to `errmsg` was unused if the condition inside the loop never evaluated to `True`. At my suggestion, @kmvanbrunt moved these lines into the loop. Yes, it looks a little strange at first, but the intent is to show the user the shortcut characters, since they can't be used at the beginning of an alias. I believe doing this any other way would result in needless string concatenation for a string that wouldn't be used in the happy-day scenario.